### PR TITLE
Ai fleet merger cleanup

### LINKED ISSUE #1794: AI exception when loading save game - assign_military_fleets_to_systems
### Environment

* **FreeOrion Version**: 0.4.7+ f5e26bc
* **Operating System**: Fedora
* **Fetched as**  Compiled from source

### Description
On load of save game posted in #1778 (from turn 129):
```
AI_4: AI_Error: AI script error : FreeOrionAI.py:generateOrders():333  - Exception 114698 while trying to assign_military_fleets_to_systems
Traceback (most recent call last):
  File "/opt/projects/freeorion/default/python/AI/FreeOrionAI.py", line 330, in generateOrders
    action()
  File "/opt/projects/freeorion/default/python/AI/MilitaryAI.py", line 803, in assign_military_fleets_to_systems
    fleet_list=found_fleets)
  File "/opt/projects/freeorion/default/python/AI/FleetUtilsAI.py", line 150, in get_fleets_for_mission
    fleet_pool_set.remove(fleet_id)
KeyError: 114698
```

### Expected Result
No exception

### PR DESCRIPTION
**Pull request by Dilvish-fo**
A bit of cleanup related to AI fleet mergers and splits.  Among other things, this appears to resolve #1794

### CODE REVIEW

**Review by Morlic-fo — COMMENTED** [review]
Some style nitpicking otherwise looks ok.

File: default/python/AI/AIFleetMission.py [hunk 1]
```diff
@@ -184,8 +189,10 @@ def check_mergers(self, context=""):
                 else:
                     target = fleet_mission.target.id if fleet_mission.target else None
                     if target == target_id:
-                        print "Military fleet %d has same target as %s fleet %d. Merging." % (fid, fleet_role, fleet_id)
-                        do_merge = True  # TODO: should probably ensure that fleetA has aggression on now
+                        print ("Military fleet %d (%d ships) has same target as %s fleet %d (%d ships). Merging former "
+                               "into latter." % (fid, len(fleet.shipIDs), fleet_role, fleet_id, len(main_fleet.shipIDs)))
```
[the remark this answers is not in the document]
Comment (Dilvish-fo) [reply default/python/AI/AIFleetMission.py]: I'd like to think it's just an old habit from before numShips was exposed, but I couldn't say for sure; regardless, I'll try to drop that habit.

**Comment by Dilvish-fo** [comment]
rebased, made the suggested changes, combined two of the more related commits for conciseness/clarity, tested and still seems to work fine, so I'll be merging

**Comment by Dilvish-fo** [comment]
minor update:  I just noticed the extra space after the ```get``` got lost when combining the two commits, so I just added that as a grooming commit to master.

**Pull request merged**